django/db/models/sql/query.py

```diff
@@ -2198,6 +2198,9 @@ def get_order_dir(field, default='ASC'):
     prefix) should sort. The '-' prefix always sorts the opposite way.
     """
     dirn = ORDER_DIR[default]
+    if isinstance(field, OrderBy):
+        return field.expression.name, 'DESC' if field.descending else 'ASC'
+
```

Comment:
I don't think that it is a proper solution because `get_order_dir()` returns only field name and direction, so it will not properly in case of using database functions in `meta.ordering`. I think we should fix this in `find_ordering_name()`, e.g.
```diff
diff --git a/django/db/models/sql/compiler.py b/django/db/models/sql/compiler.py
index a44adfc760..fb2a4b23b1 100644
--- a/django/db/models/sql/compiler.py
+++ b/django/db/models/sql/compiler.py
@@ -716,6 +716,9 @@ class SQLCompiler:
 
             results = []
             for item in opts.ordering:
+                if isinstance(item, OrderBy):
+                    results.append((item, False))
+                    continue
                 results.extend(self.find_ordering_name(item, opts, alias,
                                                        order, already_seen))
             return results
```